### PR TITLE
Changing Elastic Net Classifier params

### PR DESCRIPTION
**Pull request by bchen1116**
partial fix for #2145.
Docs [here](https://feature-labs-inc-evalml--2269.com.readthedocs.build/en/2269/tutorials.html)
Perf tests [here](https://alteryx.quip.com/DavmACoBYgBA/Elastic-Net-parameters-Perf-test-results)

While this PR does result in better performance in the docs, the main issue that still stands is that the scoring methods we had used (lead_scoring and fraud) allowed a pipeline that always guessed the positive (minority) class to have the best score. To actually solve the issue, it would be useful to tweak the objective functions so that this case would score poorly, otherwise this issue can arise again in the future.

I think that can be a separate PR that can follow some additional discussion. Things we'd need to discuss are:
- How can we create objectives that can score better, even if we are using various sampling methods?
- Can these new objectives be robust to models that only guess one class?

### CODE REVIEW

**Comment by bchen1116** [comment]
Filed an issue to track the SHAP failure [here](https://github.com/alteryx/evalml/issues/2281)

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_algorithms.py [hunk 1]
```diff
@@ -109,8 +109,12 @@ def test_shap(estimator, problem_type, n_points_to_explain, X_y_binary, X_y_mult
         is_binary = False
     else:
         training_data, y = X_y_regression
+    if "Elastic Net" in estimator.name:
+        parameters = {"Elastic Net Classifier": {"alpha": 0.5, "l1_ratio": 0.5, 'n_jobs': 1}}
```
Comment (freddyaboulton) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_algorithms.py:113]: Tracked by #2281 - @bchen1116 can you add a TODO?

**Review by freddyaboulton — APPROVED** [review]
@bchen1116 I think this looks good! Thanks for chasing down the original issue and recommending a very targeted fix. 🎉 

I agree that we can tackle the issues with custom objectives you've identified in a separate ticket.

File: docs/source/demos/fraud.ipynb [hunk 2]
```diff
@@ -68,7 +68,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "X, y = evalml.demos.load_fraud(n_rows=1000)"
+    "X, y = evalml.demos.load_fraud(n_rows=5000)"
```
Comment (bchen1116) [inline docs/source/demos/fraud.ipynb:71]: Added `n_rows=5000` as stated in the perf test doc [here](https://alteryx.quip.com/ZDhLAlkjiVR4/Fraud-and-Lead-Scoring-analysis) for performance improvement

**Review by chukarsten — COMMENTED** [review]
LGTM!

File: evalml/tests/model_understanding_tests/test_graphs.py [hunk 3]
```diff
@@ -1063,11 +1063,7 @@ def test_linear_coefficients_output(estimator):
     est_.fit(X, y)
 
     output_ = get_linear_coefficients(est_, features=['First', 'Second', 'Third', 'Fourth'])
-
-    if estimator.name == 'Linear Regressor':
-        assert (output_.index == ['Intercept', 'Second', 'Fourth', 'First', 'Third']).all()
-    elif estimator.name == 'Elastic Net Regressor':
-        assert (output_.index == ['Intercept', 'Second', 'Third', 'Fourth', 'First']).all()
+    assert (output_.index == ['Intercept', 'Second', 'Fourth', 'First', 'Third']).all()
```
Comment (ParthivNaresh) [inline evalml/tests/model_understanding_tests/test_graphs.py:1066]: Noice

**Review by ParthivNaresh — APPROVED** [review]
Nice work with the perf tests!

**Pull request merged**